backend/app/models/listeners.py

```diff
@@ -0,0 +1,88 @@
+from datetime import datetime
+from pydantic import Field, BaseModel
+from typing import Optional, List, Union
+from app.models.pyobjectid import PyObjectId
+from app.models.mongomodel import MongoModel
+from app.models.users import UserOut
+
+
+class Repository(MongoModel):
+    """Reference to a repository associated with Event Listener/Extractor."""
+
+    repository_type: str = "git"
+    repository_url: str = ""
+
+
+class ExtractorInfo(BaseModel):
+    """Currently for extractor_info JSON from Clowder v1 extractors for use with to /api/extractors endpoint."""
+
+    author: str  # Referring to author of listener script (e.g. name or email), not Clowder user
+    process: dict
+    maturity: str = "Development"
+    contributors: List[str] = []
+    contexts: List[dict] = []
+    repository: List[Repository] = []
+    external_services: List[str] = []
+    libraries: List[str] = []
+    bibtex: List[str] = []
+    default_labels: List[str] = []
+    categories: List[str] = []
+    parameters: List[dict] = []
+
+
+class EventListenerBase(BaseModel):
+    """An Event Listener is the expanded version of v1 Extractors."""
+
+    name: str
+    version: int = 1
+    description: str = ""
+
+
+class EventListenerIn(EventListenerBase):
+    """On submission, minimum info for a listener is name, version and description. Clowder will use name and version to locate queue."""
+
+    pass
+
+
+class LegacyEventListenerIn(ExtractorInfo):
+    """v1 Extractors can submit data formatted as a LegacyEventListener (i.e. v1 format) and it will be converted to a v2 EventListener."""
+
+    name: str
+    version: str = "1.0"
```

Comment:
Why the difference in types between version for `EventListenerBase` and `LegacyEventListener` (str vs int)?

Reply:
Clowder v1 lets users arbitrarily provide a version with no rules (e.g. "1", "1.0", "v1" "beta" all allowed). I was thinking (before v1 back-compatibility) that version could just be an integer for clarity and simplicity, but after seeing it is just a string field for v1 we should probably use that. 

If we use string, "version" might better be called "tag" or something like Docker does? Don't necessarily see a point in enforcing something that can be cast as a float in that case...

Reply:
> Clowder v1 lets users arbitrarily provide a version with no rules (e.g. "1", "1.0", "v1" "beta" all allowed). I was thinking (before v1 back-compatibility) that version could just be an integer for clarity and simplicity, but after seeing it is just a string field for v1 we should probably use that.
> 
> If we use string, "version" might better be called "tag" or something like Docker does? Don't necessarily see a point in enforcing something that can be cast as a float in that case...

Good point. Tag sounds great. We could also enforce semantic versioning in the string and call it version. But that might be more work. Similar to what npm does.